### PR TITLE
Niak 105

### PR DESCRIPTION
**Pull request by jfoder**

### CODE REVIEW

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public List<Hours> saveHours(final AddHoursRequest request) {
+        final Room room = roomRepository.findById(request.getRoomId())
+                .orElseThrow(() -> new RoomNotFoundException("No room with id " + request.getRoomId()));
+
+        validateIfOverlapping(request, room.getAvailabilityHours());
+
+        List<Hours> availabilityHours = conversionService.convert(request);
+        for (Hours hours : availabilityHours) {
+            hoursRepository.save(hours);
+        }
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java]: Maybe `hoursRepository.saveAll(availabilityHours);`?

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/RoomService.java [hunk 2]
```diff
@@ -59,6 +61,11 @@ public GetRoomResponse saveRoom(final AddRoomRequest request) {
         Room room = conversionService.convert(request);
         roomRepository.save(room);
 
+        if (request.getAvailabilityDetails() != null) {
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/RoomService.java:64]: `AddRoomRequest` has a `nonNull` annotation on `availabilityDetails` field.

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import org.niewidoczniakademicy.rezerwacje.service.repository.HoursRepository;
+import org.niewidoczniakademicy.rezerwacje.service.repository.RoomRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java]: Logger is not used here

File: rezerwacje-app/src/test/resources/postman/reservation-system.postman_environment.json [hunk 4]
```diff
@@ -156,9 +156,14 @@
 			"key": "connect-recruitment-and-room-path",
 			"value": "connect-recruitment-and-room",
 			"enabled": true
+		},
+		{
+			"key": "hours-path",
```
Comment (enthyp) [inline rezerwacje-app/src/test/resources/postman/reservation-system.postman_environment.json:161]: It's not relevant to this PR but some previous one, but `room-path` Postman environment variable is not defined

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                .roomId(roomId)
+                .build();
+        return saveHours(request);
+    }
+
+    public List<Hours> saveHours(final AddHoursRequest request) {
+        final Room room = roomRepository.findById(request.getRoomId())
+                .orElseThrow(() -> new RoomNotFoundException("No room with id " + request.getRoomId()));
+
+        validateIfOverlapping(request, room.getAvailabilityHours());
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java:44]: I think we should also validate if `timeStart < timeEnd` for all added TimeIntervals

**Review by enthyp — COMMENTED** [review]
Added some minor remarks, once these are fixed I think it can be merged

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java (hunk 1 above)
Comment (jfoder) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java]: Done

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/RoomService.java (hunk 2 above)
Comment (jfoder) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/RoomService.java:64]: Removed `@NonNull` annotation from `availabilityDetails` as this field is not required

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java (hunk 3 above)
Comment (jfoder) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java]: Removed

File: rezerwacje-app/src/test/resources/postman/reservation-system.postman_environment.json (hunk 4 above)
Comment (jfoder) [reply rezerwacje-app/src/test/resources/postman/reservation-system.postman_environment.json:161]: Added

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java (hunk 5 above)
Comment (jfoder) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java:44]: I forgot about that, added

**Pull request merged**